### PR TITLE
Add dataservices search page

### PR DESCRIPTION
**Pull request by ThibaudDauce**
Require https://github.com/opendatateam/udata/pull/3148

Fix https://github.com/datagouv/data.gouv.fr/issues/1471

### CODE REVIEW

File: udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+                    :isBlue="true"
+                  />
+                </div>
+                <div class="fr-col-12">
+                  <div class="fr-select-group">
+                    <label class="fr-label" :for="isRestrictedId">
+                      {{ t("Access terms") }}
+                    </label>
+                    <select class="fr-select" :id="isRestrictedId" v-model="isRestricted">
+                      <option :value="null" selected disabled hidden>Sélectionner une option</option>
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue:58]: Missing translation

Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue:58]: Also, don't we want this option in the list ? Other filters have an `all` option

File: udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+                    @change="(value: string) => organization = value"
+                    :isBlue="true"
+                  />
+                </div>
+                <div class="fr-col-12">
+                  <div class="fr-select-group">
+                    <label class="fr-label" :for="isRestrictedId">
+                      {{ t("Access terms") }}
+                    </label>
+                    <select class="fr-select" :id="isRestrictedId" v-model="isRestricted">
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue:57]: Changes to this filter should all reset the page to 1

File: udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+  window.history.pushState(null, "", url);
+
+  return `/dataservices?${params}`
+})
+
+const dataservices = ref<null | PaginatedArray<Dataservice>>(null);
+
+const search = async () => {
+  dataservices.value = null
+  const response = await api.get(url.value);
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue:187]: You can add an [AbortController from axios](https://axios-http.com/docs/cancellation) to cancel unwanted request. The helper already exists in `udata_front/theme/gouvfr/datagouv-components/src/api/api.ts`.

File: udata_front/theme/gouvfr/assets/js/locales/en.json [hunk 4]
```diff
@@ -483,5 +483,9 @@
   "Member since": "Member since",
   "Last connection": "Last connection",
   "No connection": "No connection",
-  "You are seeing a specific community resource of this dataset": "You are seeing a specific community resource of this dataset"
+  "You are seeing a specific community resource of this dataset": "You are seeing a specific community resource of this dataset",
+  "Access terms": "Access terms",
+  "Open APIs to everyone": "Open APIs to everyone",
+  "Restricted access APIs": "Restricted access APIs",
+  "{count} API": "{count} API"
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/locales/en.json:490]: Do we want to show plurals for this one ?

**Review by nicolaskempf57 — COMMENTED** [review]
Thanks ! Some minor changes and this is ready 🙏🏻

File: udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+});
+const resetFilters = () => {
+  organization.value = '';
+  isRestricted.value = null;
+}
+
+const searchId = useId();
+const searchQuery = ref('');
+const searchQueryDebounced = refDebounced(searchQuery, 500);
+const searchInput = useTemplateRef('searchInput');
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue:158]: Can you rename this or update vue to 3.5.6 ? There is a [bug](https://github.com/vuejs/core/issues/11795#issuecomment-2326858438) that cause a warning because the variable name is the same as the ref

File: udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue (hunk 1 above)
Comment (ThibaudDauce) [reply udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue:58]: This is the all option, (passing `null` will return both private and public APIs). Is it fine with you? Or should we be more explicit with an "All API option"?

File: udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue (hunk 3 above)
Comment (ThibaudDauce) [reply udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue:187]: What is the point of the abort controller? Is it really useful in this case with a debounce of 500ms?

File: udata_front/theme/gouvfr/assets/js/locales/en.json (hunk 4 above)
Comment (ThibaudDauce) [reply udata_front/theme/gouvfr/assets/js/locales/en.json:490]: There is no plural in the Figma but we could add it…

File: udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue [hunk 6]
```diff
@@ -0,0 +1,194 @@
+<template>
+  <form class="fr-pt-3v" @submit.prevent="search">
+    <div class="fr-grid-row fr-grid-row--middle justify-between" ref="searchRef" data-cy="search">
```
[the remark this answers is not in the document]
Comment (ThibaudDauce) [reply udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue]: Done in 92e2b8ac

File: udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue (hunk 2 above)
Comment (ThibaudDauce) [reply udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue:57]: Done in 92e2b8ac

File: udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue (hunk 5 above)
Comment (ThibaudDauce) [reply udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue:162]: Done in 92e2b8ac

File: udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue (hunk 1 above)
Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue]: The main issue is that once another option is selected, this one can't be picked from the list. Also We could rephrase it to "All access term" or something to match the wording of the organization field

File: udata_front/theme/gouvfr/assets/js/locales/en.json (hunk 4 above)
Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/assets/js/locales/en.json:490]: Let's keep it without plural then

File: udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue (hunk 1 above)
Comment (ThibaudDauce) [reply udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue]: Good catch! I didn't thought of unselecting :-) Done in 7905c904

File: udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue (hunk 3 above)
Comment (ThibaudDauce) [reply udata_front/theme/gouvfr/assets/js/components/DataservicesSearch/DataservicesSearch.vue]: Done :-)

**Review by nicolaskempf57 — APPROVED** [review]
